From 1509a9b466b7793c19b833de073748a33b71b35a Mon Sep 17 00:00:00 2001
From: "Bernhard M. Wiedemann" <bwiedemann@suse.de>
Date: Tue, 15 Aug 2017 22:29:07 +0200
Subject: [PATCH] Sort generated entries

When building packages (e.g. for openSUSE Linux), varying ordering of
functions in the output would cause differing binaries.

See https://reproducible-builds.org/ for why this matters.

Change-Id: I29a2e54db766fba8a9cd3e5f1a1e552e603a7788
Reviewed-by: Oswald Buddenhagen <oswald.buddenhagen@qt.io>
---
 generator/setupgenerator.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/generator/setupgenerator.cpp b/generator/setupgenerator.cpp
index 25a98ea..6f176f8 100644
--- generator/setupgenerator.cpp
+++ generator/setupgenerator.cpp
@@ -64,6 +64,7 @@ void SetupGenerator::generate()
         QList<const AbstractMetaClass*> list = pack.value();
         if (list.isEmpty())
             continue;
+        std::sort(list.begin(), list.end());
 
         QString packName = pack.key();
         QStringList components = packName.split(".");
